addons/mail/models/mail_activity.py

```diff
@@ -179,29 +179,23 @@ def _onchange_activity_type_id(self):
         if self.activity_type_id:
             if self.activity_type_id.summary:
                 self.summary = self.activity_type_id.summary
-            self.date_deadline = self._calculate_date_deadline(self.activity_type_id)
+            self.date_deadline = self._get_date_deadline(self.activity_type_id)
             self.user_id = self.activity_type_id.default_user_id or self.env.user
             if self.activity_type_id.default_note:
                 self.note = self.activity_type_id.default_note
 
     @api.model
-    def _calculate_date_deadline(self, activity_type, force_base_date=None):
-        """ Compute the activity deadline given its type, the force_base_date and the context.
+    def _get_date_deadline(self, activity_type):
```

Comment:
Sorry, maybe that was unclear, I meant moving this method to the "mail.activity.type" model and make it record based.

So you would call: `activity_type._get_date_deadline()` instead of `env['mail.activity']._get_date_deadline(activity_type)`.